test/middleware/AuthMiddleware.test.ts

```diff
@@ -0,0 +1,127 @@
+import 'reflect-metadata';
+import '../../src/controllers/index';
+import { expect, assert } from 'chai';
+import * as request from 'supertest';
+import { PENALTY_DETAILS_PAGE_URI, OTHER_REASON_DISCLAIMER_PAGE_URI, OTHER_REASON_PAGE_URI } from '../../src/utils/Paths';
+import { createApp, getDefaultConfig } from '../ApplicationFactory';
+import { Maybe, VerifiedSession } from 'ch-node-session-handler';
+import { createFakeSession } from '../utils/session/FakeSessionFactory';
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { generateSessionId, generateSignature } from 'ch-node-session-handler/lib/utils/CookieUtils';
+
+const config = getDefaultConfig();
+const id = generateSessionId();
+const sig = generateSignature(id, config.cookieSecret);
+
+const session = createFakeSession(
+    [{
+        [SessionKey.Id]: id
+    }, {
+        [SessionKey.ClientSig]: sig
+    }], config.cookieSecret, true);
+
+const authedApp = createApp(session);
+
+const protectedPages = [
+    PENALTY_DETAILS_PAGE_URI,
+    OTHER_REASON_DISCLAIMER_PAGE_URI,
+    OTHER_REASON_PAGE_URI
+];
+
+describe('Authentication Middleware', () => {
+
+
+    describe('Authed path', () => {
+
+        let penaltyDetails: Record<string, any>;
+
+        before('sign in server should be running', async () => {
+
+            penaltyDetails = {
+                penaltyReference: 'A12345678',
+                companyNumber: 'SC123123'
+            };
+
+            await request(createApp())
+                .get(PENALTY_DETAILS_PAGE_URI)
+                .catch(err => assert.fail(err));
+
+
+        });
+
+        it('should not redirect the user to the sign in page if the user is signed in', async () => {
+
+            for (const page of protectedPages) {
+                await request(authedApp).get(page)
+                    .expect(200);
+            }
+
+        });
+
+        it('should allow the user to go to the penalty reference screen if authed', async () => {
+
+            await request(authedApp)
+                .get(PENALTY_DETAILS_PAGE_URI)
+                .expect(200);
+
+        });
+        it('should continue to the next page without requiring auth if penalty details are valid', async () => {
+
+            await request(authedApp)
+                .post(PENALTY_DETAILS_PAGE_URI)
+                .send(penaltyDetails)
+                .expect(_ => expect(_.header.location).to.include(OTHER_REASON_DISCLAIMER_PAGE_URI));
+
+        });
+    });
+    describe('Unauthed path', () => {
+
+        const appWithoutCookie = { label: 'App without Cookie', app: createApp(session, false) };
```

Comment:
Two things:
1) Auth middleware is not concerned with cookie.
2)` setCookie` in `const createApp = (session?: Session, setCookie: boolean = true)` is never used - likely because of 1)

I suggest you remove it due to above reasons.